bam/writer.go

```diff
@@ -53,7 +53,10 @@ func NewWriterLevel(w io.Writer, h *sam.Header, level, wc int) (*Writer, error)
 	if err != nil {
 		return nil, err
 	}
-	bw.bg.Flush()
+	err = bw.bg.Flush()
```

Comment:
This is not necessary. `error`s in `bgzf.Writer` are sticky, so any failure is caught at the call to `Wait` that follows.